test/qa/test_dns.py

```diff
@@ -25,92 +24,73 @@ def teardown_module(module):
 def setup_function(function):
     logging.log()
 
+    # Make sure that Custom DNS, IPv6 and Threat Protection Lite are disabled before we execute each test
+    lib.set_dns("off")
+    lib.set_ipv6("off")
+    lib.set_threat_protection_lite("off")
+
 
 def teardown_function(function):
     logging.log(data=info.collect())
     logging.log()
 
 
 @pytest.mark.parametrize("threat_protection_lite", lib.THREAT_PROTECTION_LITE)
-@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES_BASIC1)
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES)
 @pytest.mark.flaky(reruns=2, reruns_delay=90)
 @timeout_decorator.timeout(40)
 def test_dns_connect(tech, proto, obfuscated, threat_protection_lite):
     lib.set_technology_and_protocol(tech, proto, obfuscated)
     lib.set_threat_protection_lite(threat_protection_lite)
 
+    # Make sure, that DNS is unset before we connect to VPN server
     assert dns.is_unset()
 
-    output = sh.nordvpn.connect()
-
-    print(output)
-    assert lib.is_connect_successful(output)
+    with lib.Defer(sh.nordvpn.disconnect):
+        sh.nordvpn.connect()
+        if threat_protection_lite == "on":
+            assert dns.is_set_for(dns.DNS_TPL)
+        else:
+            assert dns.is_set_for(dns.DNS_NORD)
 
-    with lib.ErrorDefer(sh.nordvpn.disconnect):
-        assert network.is_connected()
-        assert dns.is_set_for(threat_protection_lite)
-
-    output = sh.nordvpn.disconnect()
-    print(output)
-    assert lib.is_disconnect_successful(output)
-    assert network.is_disconnected()
+    # Make sure, that DNS is unset, after we disconnect from VPN server
     assert dns.is_unset()
 
 
-@pytest.mark.parametrize("nameserver", lib.DNS)
-@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES_BASIC1)
+@pytest.mark.parametrize("nameserver", dns.DNS_CASES_CUSTOM)
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES)
 @pytest.mark.flaky(reruns=2, reruns_delay=90)
 @timeout_decorator.timeout(40)
 def test_custom_dns_connect(tech, proto, obfuscated, nameserver):
+    nameserver = nameserver.split(" ")
+
     lib.set_technology_and_protocol(tech, proto, obfuscated)
     sh.nordvpn.set.dns(nameserver)
 
     assert dns.is_unset()
 
-    output = sh.nordvpn.connect()
-    print(output)
-    assert lib.is_connect_successful(output)
-
-    with lib.ErrorDefer(sh.nordvpn.disconnect):
-        assert network.is_connected()
-        assert dns.is_unset()
+    with lib.Defer(sh.nordvpn.disconnect):
+        sh.nordvpn.connect()
 
-    output = sh.nordvpn.disconnect()
-    print(output)
-    assert lib.is_disconnect_successful(output)
-    assert network.is_disconnected()
+        assert dns.is_set_for(nameserver)
 
     sh.nordvpn.set.dns.off()
```

Comment:
What should the DNS be set to after disconnect but before setting DNS off? Should add an additional assert. Unless it is also unset, in which case `sh.nordvpn.set.dns.off()` wouldn't be necessary then.

Reply:
Since we unset DNS before executing each test, in `setup_function`, I think we can remove the `sh.nordvpn.set.dns.off()`